feature/experimental/interface/loopback/ate_tests/loopback_aggregate_test/interface_loopback_aggregate_test.go

```diff
@@ -0,0 +1,225 @@
+		t.Errorf("%s LagID got %v, want %v", dp, lagID, aggID)
+	}
+}
+
+// configureDUTPort1 configures port1 on DUT.
+func configureDUTPort1(t *testing.T, dut *ondatra.DUTDevice, dutOcRoot *oc.Root, dutPort1 *ondatra.Port) {
+	intf := dutOcRoot.GetOrCreateInterface(dutPort1.Name())
+	intf.Type = ethernetCsmacd
+	if deviations.InterfaceEnabled(dut) {
+		intf.Enabled = ygot.Bool(true)
+	}
+
+	dutOcPath := gnmi.OC()
+	fptest.LogQuery(t, fmt.Sprintf("%s to Update()", dut), dutOcPath.Config(), dutOcRoot)
+	gnmi.Update(t, dut, dutOcPath.Config(), dutOcRoot)
+}
+
+// configureDUT configures AE interface and adds port1 to AE.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice, dutOcRoot *oc.Root, aggID string, dutPort1 *ondatra.Port) {
+	agg := dutOcRoot.GetOrCreateInterface(aggID)
+	agg.GetOrCreateAggregation().LagType = lagType
+	agg.Type = ieee8023adLag
+
+	intf := dutOcRoot.GetOrCreateInterface(dutPort1.Name())
+	intf.GetOrCreateEthernet().AggregateId = ygot.String(aggID)
+	intf.Type = ethernetCsmacd
+
+	if deviations.InterfaceEnabled(dut) {
+		intf.Enabled = ygot.Bool(true)
+	}
+
+	dutOcPath := gnmi.OC()
+	fptest.LogQuery(t, fmt.Sprintf("%s to Update()", dut), dutOcPath.Config(), dutOcRoot)
+	gnmi.Update(t, dut, dutOcPath.Config(), dutOcRoot)
+
+	aggIntf := &oc.Interface{Name: ygot.String(aggID)}
+	configAggregateIntf(dut, aggIntf, &dutPort1Attr)
+	aggPath := dutOcPath.Interface(aggID)
+	fptest.LogQuery(t, aggID, aggPath.Config(), aggIntf)
+	gnmi.Replace(t, dut, aggPath.Config(), aggIntf)
+}
+
+// configureATE configures port1 on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) {
+	topo := ate.Topology().New()
+	port1 := ate.Port(t, "port1")
+	topo.AddInterface(atePort1Attr.Name).WithPort(port1)
+	topo.Push(t).StartProtocols(t)
+}
+
+// configSrcDUT configures AE interface.
+func configSrcDUT(dut *ondatra.DUTDevice, i *oc.Interface, a *attrs.Attributes) {
+	i.Description = ygot.String(a.Desc)
+	if deviations.InterfaceEnabled(dut) {
+		i.Enabled = ygot.Bool(true)
+	}
+
+	s := i.GetOrCreateSubinterface(0)
+	s4 := s.GetOrCreateIpv4()
+	if deviations.InterfaceEnabled(dut) && !deviations.IPv4MissingEnabled(dut) {
+		s4.Enabled = ygot.Bool(true)
+	}
+	a4 := s4.GetOrCreateAddress(a.IPv4)
+	a4.PrefixLength = ygot.Uint8(plenIPv4)
+
+	s6 := s.GetOrCreateIpv6()
+	if deviations.InterfaceEnabled(dut) {
+		s6.Enabled = ygot.Bool(true)
+	}
+	s6.GetOrCreateAddress(a.IPv6).PrefixLength = ygot.Uint8(plenIPv6)
+}
+
+// configAggregateIntf configures AE interface on DUT.
+func configAggregateIntf(dut *ondatra.DUTDevice, i *oc.Interface, a *attrs.Attributes) {
+	configSrcDUT(dut, i, a)
+	i.Type = ieee8023adLag
+	g := i.GetOrCreateAggregation()
+	g.LagType = lagType
+}
+
+// TestInterfaceLoopbackMode is to test loopback mode FACILITY.
+func TestInterfaceLoopbackMode(t *testing.T) {
+	t.Logf("Start DUT config load.")
+	dut := ondatra.DUT(t, "dut")
+	dutOcRoot := &oc.Root{}
+	aggID := netutil.NextBundleInterface(t, dut)
```

Comment:
NextBundleInterface is deprecated. Please use NextAggregateInterface instead.

Reply:
updated, thank you